zeroconf/__init__.py

```diff
@@ -1423,14 +1423,22 @@ def __init__(self, zc: 'Zeroconf') -> None:
 
     def run(self) -> None:
         while True:
-            self.zc.wait(10 * 1000)
+            self.zc.wait(60 * 1000)
             if self.zc.done:
                 return
             now = current_time_millis()
-            for record in self.zc.cache.entries():
-                if record.is_expired(now):
-                    self.zc.update_record(now, record)
-                    self.zc.cache.remove(record)
+            # Only copy the values that are going to expire
+            # instead of copying all the values as the cache
+            # can be very large if the LAN has a signficant
+            # number of devices generating mDNS traffic.
+            remove = [
+                record
+                for record in itertools.chain.from_iterable(self.zc.cache.cache.values())
```

Comment:
I left this as a draft as I need to verify we won't trigger a `dictionary changed size during iteration` error